src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,81 @@
 package core.basesyntax;
 
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.EmployeeWorkDayInfo;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
 public class SalaryInfo {
+    private static final int INDEX_OF_DATES = 0;
+    private static final int INDEX_OF_NAMES = 1;
+    private static final int INDEX_OF_HOURS = 2;
+    private static final int INDEX_OF_WAGES = 3;
+    private static final DateTimeFormatter TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        List<EmployeeWorkDayInfo> employeeWorkDayInfoList = parseData(data);
+        filterByDate(employeeWorkDayInfoList, dateFrom, dateTo);
+        Queue<String> statistic = calculate(names);
+        Storage.getEmployeeWorkDayInfo().clear();
+        return makeReport(statistic, names, dateFrom, dateTo);
+    }
+
+    public String makeReport(Queue<String> statistic, String[] names,
+                             String dateFrom, String dateTo) {
+        String header = "Report for period " + dateFrom + " - " + dateTo + System.lineSeparator();
+        StringBuilder builder = new StringBuilder(header);
+        Arrays.stream(names).forEach(name -> {
+            builder.append(statistic.poll());
+            if (statistic.peek() != null) {
+                builder.append(System.lineSeparator());
+            }
+        });
+        return builder.toString();
+    }
+
+    public Queue<String> calculate(String[] names) {
+        Queue<String> salaryStatistic = new LinkedList<>();
+        Arrays.stream(names).forEach(name -> {
+            Integer salary = Storage.getEmployeeWorkDayInfo().stream()
+                    .filter(employeeWorkDayInfo ->
+                            employeeWorkDayInfo.getEmployeeName().equals(name))
+                    .map(employeeWorkDayInfo -> employeeWorkDayInfo.getHours()
+                            * employeeWorkDayInfo.getIncomePerHour())
+                    .reduce(0, Integer::sum);
+            String employeeSalary = name + " - " + salary;
```

Comment:
again concatenation, let's rewrite your logic, so you won't need to use Queue or think about how don't use concatenation here